Library/Homebrew/cmd/cleanup.rb

```diff
@@ -62,6 +62,10 @@ def cleanup_cache
         cleanup_path(path) { path.unlink }
         next
       end
+      if path.to_s.include?("java_cache")
```

Comment:
Simple example: If `HOMEBREW_CACHE` happens to be `/Users/java_cache/homebrew/cache` (I know that `java_cache` is a pretty stupid user name) then this will wipe out the entire cache every time `brew cleanup` is invoked.

> I'm perfectly happy to make the cache name a whole lot more random if desired.

Please let's not go down that route. I don't see why we shouldn't use a 100% safe and simple alternative instead of leaving a nonzero (if small) chance of things going completely wrong.